src/components/chips/js/chipsDirective.js

```diff
@@ -202,13 +202,18 @@ function MdChips ($mdTheming, $mdUtil, $compile, $timeout) {
 
     // Set the chip remove, chip contents and chip input templates. The link function will put
     // them on the scope for transclusion later.
-    var chipRemoveTemplate   = getTemplateByQuery('[md-chip-remove]') || CHIP_REMOVE_TEMPLATE,
-        chipContentsTemplate = getTemplateByQuery('md-chip-template') || CHIP_DEFAULT_TEMPLATE,
-        chipInputTemplate    = getTemplateByQuery('md-autocomplete')
-            || getTemplateByQuery('input')
+    var chipRemoveTemplate   = getTemplateByQuery('md-chips>*[md-chip-remove]') || CHIP_REMOVE_TEMPLATE,
+        chipContentsTemplate = getTemplateByQuery('md-chips>md-chip-template') || CHIP_DEFAULT_TEMPLATE,
+        chipInputTemplate    = getTemplateByQuery('md-chips>md-autocomplete')
+            || getTemplateByQuery('md-chips>input')
             || CHIP_INPUT_TEMPLATE,
         staticChips = userTemplate.find('md-chip');
 
+    // Warn of malformed template. See #2545
+    if (userTemplate[0].querySelector('md-chip-template>*[md-chip-remove]')) {
+      $log.warn('invalid placement of md-chip-remove within md-chip-template.');
+    }
+
```

Comment:
I like this - and I should start adding more obvious warnings for known cases of confusion.  Noted. =p